src/backend/base/langflow/components/nvidia/nemo_evaluator.py

```diff
@@ -382,22 +489,220 @@ def _generate_custom_evaluation_body(self) -> dict:
+            },
+            "evaluations": [
+                {
+                    "eval_type": "llm_as_a_judge",
+                    "eval_subtype": "mtbench",
+                    "bench_name": "mt_bench",
+                    "mode": "single",
+                    "evaluation_stages": getattr(self, "evaluation_stages", ["generation", "judgement"]),
+                    "inference_params": inference_params,
+                    "judge_model": judge_model,
+                    "judge_inference_params": judge_inference_params,
                 }
             ],
             "tag": getattr(self, "001_tag", ""),
         }
+
+    def get_dataset_name(self, user_dataset_name=None):
+        # Generate a default dataset name using the current date and time
+        default_name = f"dataset-{datetime.now(timezone.utc).strftime('%Y%m%d%H%M%S')}"
+        # Use the user-provided name if available, otherwise the default
+        return user_dataset_name if user_dataset_name else default_name
+
+    async def get_dataset_id(self, dataset_name: str) -> str:
+        """Fetches the dataset ID by checking if a dataset with the constructed name exists.
+
+        If the dataset does not exist, creates a new dataset and returns its ID.
+
+        Args:
+            dataset_name (str): Provided dataset name.
+
+        Returns:
+            str: The dataset ID if found or created, or None if an error occurs.
+        """
+
+        url = f"{self.datastore_base_url}/v1/datasets"
+        page = 1
+
+        try:
+            async with httpx.AsyncClient() as client:
+                while True:
+                    response = await client.get(f"{url}?page_size=10&page={page}")
+                    no_more_pages = 422
+                    if response.status_code == no_more_pages:
+                        self.log(f"No more pages to fetch, page: {page}")
+                        break
+                    response.raise_for_status()
+                    self.log(f"returned data {response}")
+                    datasets = response.json().get("datasets", [])
+                    for dataset in datasets:
+                        if dataset.get("name") == dataset_name:
+                            return dataset.get("id")
+                    if not datasets:
+                        break  # No more datasets to process
+
+                    page += 1
+
+                # If dataset not found, create it
+                create_payload = {"name": dataset_name, "description": f"{dataset_name} for Nemo custom eval"}
+                create_response = await client.post(url, json=create_payload)
+                create_response.raise_for_status()
+                created_dataset = create_response.json()
+                return created_dataset.get("id")
+
+        except httpx.HTTPStatusError as e:
+            self.log(f"Error processing datasets: {e}")
+            return None
+
+    async def process_and_upload_dataset(self) -> str:
```

Comment:
Guessing that this method will create a new dataset every time it is invoked. Do we want to be able to reuse the same dataset if the component is re-run with the same input?

Reply:
If user provides a dataset name, it will reuse same dataset in the datastore.